src/execution/join_hashtable.cpp

```diff
@@ -794,4 +888,221 @@ idx_t JoinHashTable::FillWithHTOffsets(data_ptr_t *key_locations, JoinHTScanStat
 	}
 	return key_count;
 }
+
+void JoinHashTable::SwizzleBlocks() {
+	RowDataCollectionScanner::AlignHeapBlocks(*swizzled_block_collection, *swizzled_string_heap, *block_collection,
```

Comment:
Make sure you have my 2022-08-17 changes in b8de547cf6 here - the swizzling was not right across block boundaries.  Which I have found is the really dangerous part of all this stuff...

Reply:
Gotcha. I think we fixed the code for `AlignHeapBlocks` now, especially with your additional efforts to validate the swizzling code.